ppl/cmd/zar/index/ls.go

```diff
@@ -0,0 +1,145 @@
+package index
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/brimsec/zq/cli/outputflags"
+	"github.com/brimsec/zq/ppl/archive"
+	"github.com/brimsec/zq/ppl/archive/index"
+	"github.com/brimsec/zq/ppl/cmd/zar/root"
+	"github.com/brimsec/zq/zbuf"
+	"github.com/mccanne/charm"
+	"github.com/segmentio/ksuid"
+)
+
+var Ls = &charm.Spec{
+	Name:  "ls",
+	Usage: "ls [-R root] [options]",
+	Short: "list and display stats for indices defined in archive",
+	New:   NewLs,
+}
+
+type LsCommand struct {
+	*root.Command
+
+	output outputflags.Flags
+	root   string
+	stats  bool
+}
+
+func NewLs(parent charm.Command, f *flag.FlagSet) (charm.Command, error) {
+	c := &LsCommand{Command: parent.(*Command).Command}
+	f.StringVar(&c.root, "R", os.Getenv("ZAR_ROOT"), "root location of zar archive to walk")
+	f.BoolVar(&c.stats, "stats", false, "print stats for each index definition")
+	c.output.SetSimpleFlags(f)
+	c.output.Format = "table"
+	return c, nil
+}
+
+type DefLine struct {
+	ID    string `zng:"id"`
+	Desc  string `zng:"desc"`
+	ZQL   string `zng:"zql"`
+	Input string `zng:"input"`
+}
+
+type DefStatLine struct {
+	ID         string `zng:"id"`
+	Desc       string `zng:"desc"`
+	ZQL        string `zng:"zql"`
+	Input      string `zng:"input"`
+	IndexCount uint64 `zng:"index_count"`
+	ChunkCount uint64 `zng:"chunk_count"`
+}
+
+func (c *LsCommand) Run(args []string) error {
+	defer c.Cleanup()
+	if err := c.Init(&c.output); err != nil {
+		return err
+	}
+	if c.root == "" {
+		return errors.New("zar index ls: a directory must be specified with -R or ZAR_ROOT")
+	}
+
+	ark, err := archive.OpenArchive(c.root, nil)
+	if err != nil {
+		return fmt.Errorf("zar index ls: %w", err)
+	}
+
+	defs, err := ark.ReadDefinitions(context.TODO())
+	if err != nil {
+		return fmt.Errorf("zar index ls: %w", err)
+	}
+
+	w, err := c.output.Open()
+	if err != nil {
+		return fmt.Errorf("zar indexdefs: %w", err)
```

Comment:
```suggestion
		return fmt.Errorf("zar index ls: %w", err)
```

Reply:
Would be nice if charm did this automatically.